### PR TITLE
David/referral more tests

### PR DESCRIPTION
**Pull request by WyseNynja**

### CODE REVIEW

File: migration/src/m20230708_151756_rpc_accounting_free_usage_credits.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .alter_table(
+                Table::alter()
+                    .table(RpcAccountingV2::Table)
+                    .add_column(
+                        ColumnDef::new(RpcAccountingV2::SumInclFreeCreditsUsed)
+                            .decimal_len(20, 10)
+                            .not_null(),
```
Comment (WyseNynja) [inline migration/src/m20230708_151756_rpc_accounting_free_usage_credits.rs:16]: should this have `.default(0)`?

File: migration/src/m20230708_152131_referral_track_one_time_bonus_bonus.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .alter_table(
+                Table::alter()
+                    .table(Referee::Table)
+                    .drop_column(Referee::CreditsAppliedForReferee)
+                    .add_column(
+                        ColumnDef::new(Referee::OneTimeBonusAppliedForReferee)
+                            .decimal_len(20, 10)
+                            .not_null(),
```
Comment (WyseNynja) [inline migration/src/m20230708_152131_referral_track_one_time_bonus_bonus.rs:17]: as above, should this have `.default(0)`?

File: migration/src/m20230708_225009_balance_view_v2.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        // Replace the sample below with your own migration scripts
+        todo!();
```
Comment (WyseNynja) [inline migration/src/m20230708_225009_balance_view_v2.rs:10]: https://discord.com/channels/873880840487206962/900758302294704188/1127456562865258548

from my searching, it looks like sea_orm doesn't have any official support for Views. This means their tooling won't create the entities for us or help us write the migrations.

Given that it is a read only table and we will just be doing a simple `.filter(User::Id.eq(user_id)` on it, i think we should just add the query select logic to the existing `Balance` model. Given the view isn't a "materialized view", I wonder if we should even bother. We can just write the sql in a  `async fn for_user(id: u64) -> Balance`

File: migration/src/m20230708_151756_rpc_accounting_free_usage_credits.rs (hunk 1 above)
Comment (yenicelik) [reply migration/src/m20230708_151756_rpc_accounting_free_usage_credits.rs:16]: another inconsistency i found in the table; sometimes its marked as .default("0.0"), .default(0.0), and one time not at all

File: migration/src/m20230708_225009_balance_view_v2.rs (hunk 3 above)
Comment (yenicelik) [reply migration/src/m20230708_225009_balance_view_v2.rs:10]: where would you want to introduce this function?

**Pull request closed without merging**